Exiled.API/Features/Npc.cs

```diff
@@ -196,5 +213,125 @@ public void Destroy()
             Dictionary.Remove(GameObject);
             Object.Destroy(GameObject);
         }
+
+        /// <summary>
+        /// Makes the NPC look at the specified position.
+        /// </summary>
+        /// <param name="position">The position to look at.</param>
+        public void LookAt(Vector3 position)
+        {
+            if (RoleManager.CurrentRole is IFpcRole fpc)
+                fpc.LookAtPoint(position);
+        }
+
+        /// <summary>
+        /// Makes the NPC Shoot at the specified location if holding a <see cref="Firearm"></see>.
+        /// </summary>
+        /// <param name="targetPos">The position to shoot towards.</param>
+        public void Shoot(Vector3? targetPos = null)
+        {
+            if (RoleManager.CurrentRole is not IFpcRole fpc)
+                return;
+
+            Firearm? firearm = ReferenceHub.inventory._curInstance as Firearm;
+            if (firearm == null)
+                return;
+
+            if (targetPos != null)
+            {
+                LookAt((Vector3)targetPos);
+            }
+
+            ShotMessage message = new ShotMessage()
+            {
+                ShooterCameraRotation = CameraTransform.rotation,
+                ShooterPosition = new RelativePosition(Transform.position),
+                ShooterWeaponSerial = CurrentItem.Serial,
+                TargetNetId = 0,
+                TargetPosition = default,
+                TargetRotation = Quaternion.identity,
+            };
+
+            Physics.Raycast(CameraTransform.position, CameraTransform.forward, out RaycastHit hit, 100f, StandardHitregBase.HitregMask);
+
+            if (hit.transform && hit.transform.TryGetComponentInParent(out NetworkIdentity networkIdentity) && networkIdentity)
```

Comment:
I think instead of `NetworkIdentety` here should be `IDestructable`. Iirc base game shooting logic works like this